x-pack/plugins/observability/public/components/app/observability_status/observability_status_box.tsx

```diff
@@ -0,0 +1,123 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface Props {
+  dataSource: string;
+  hasData: boolean;
+  description: string;
+  modules: Array<{ name: string; hasData: boolean }>;
+  integrationLink: string;
+  learnMoreLink: string;
+}
+
+export function ObservabilityStatusBox(props: Props) {
+  if (props.hasData) {
+    return <CompletedStatusBox {...props} />;
+  } else {
+    return <EmptyStatusBox {...props} />;
+  }
+}
+
+export function CompletedStatusBox({ dataSource, modules, integrationLink }: Props) {
+  return (
+    <EuiPanel color="subdued" style={{ marginBottom: 20 }}>
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem>
+          <div>
+            <EuiIcon
+              type="checkInCircleFilled"
+              color="success"
+              className="eui-displayInline eui-alignMiddle"
+              style={{ marginRight: 8 }}
+            />
+            <EuiTitle size="xs" className="eui-displayInline eui-alignMiddle">
+              <h2>{dataSource}</h2>
+            </EuiTitle>
+          </div>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty size="xs" iconType="plusInCircle" flush="right" href={integrationLink}>
+            <FormattedMessage
+              id="xpack.observability.status.addIntegrationLink"
+              defaultMessage="Add"
+            />
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiFlexGroup alignItems="baseline" gutterSize="s" style={{ marginTop: 8 }} role="list">
+        {modules.map((module) => (
+          <EuiFlexItem role="listitem" key={module.name}>
+            <EuiBadge
+              color={module.hasData ? 'success' : 'hollow'}
+              iconType="check"
+              iconSide="left"
+            >
+              {module.name}
+            </EuiBadge>
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+}
+
+export function EmptyStatusBox({ dataSource, description, learnMoreLink }: Props) {
+  return (
+    <EuiPanel color="subdued" style={{ marginBottom: 20 }}>
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem>
+          <div>
+            <EuiIcon
+              type="dot"
+              color="warning"
+              className="eui-displayInline eui-alignMiddle"
+              style={{ marginRight: 8 }}
+            />
+            <EuiTitle size="xs" className="eui-displayInline eui-alignMiddle">
+              <h2>{dataSource}</h2>
+            </EuiTitle>
+          </div>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiIcon type="cross" color="text" />
+        </EuiFlexItem>
```

Comment:
I see a discrepancy between the `Empty` and the `Complete` boxes here. 

The `Empty` box has this close button, whereas the `Complete` has an "Add" button. Would it make sense to have both options in both boxes?

/cc @formgeist

Reply:
You are right, I tried to build it as close as the design I could. I didn't add any behavior to the close button because I think we don't know what it should do. 
My idea was to iterate it in future versions and users won't see it until the feature toggle is on, so I think it's safe to keep it like this.